apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts

```diff
@@ -280,7 +348,14 @@ export class TypeScriptBuilder extends SubprocessRunnerBase<ITypeScriptBuilderCo
       () => {
         this._overrideTypeScriptReadJson(ts);
         const _tsconfig: TTypescript.ParsedCommandLine = this._loadTsconfig(ts);
-        const _compilerHost: TTypescript.CompilerHost = this._buildIncrementalCompilerHost(ts, _tsconfig);
+
+        let _compilerHost: TTypescript.CompilerHost;
+        if (this._capabilities.incrementalProgram) {
+          _compilerHost = this._buildIncrementalCompilerHost(ts, _tsconfig);
+        } else {
+          _compilerHost = ts.createCompilerHost(_tsconfig.options);
```

Comment:
`_buildIncrementalCompilerHost` replaces some of the filesystem APIs the compiler uses with our cached version. We should probably do the same for the non-incremental version.

Reply:
Good catch!  I didn't realize "incremental compiler" was a separate optimization from "incremental program."  I'll put that back.